### PR TITLE
refactor(accordion): fix older iOS ResizeObserver compatibility issue PCH-501

### PR DESCRIPTION
**Pull request by eeeeean**
## Jira

[DS: Fix AccordionDetails for Old iOS Versions](https://littlespoon.atlassian.net/browse/PCH-501)

## Test Plan
- Should throw an error when visiting [current accordion storybook](https://littlespoon.netlify.app/?path=/story/design-system-accordion--multiple) with iOS 13.3 or lower
- Should load when visiting [PR's accordion storybook](https://deploy-preview-817--littlespoon.netlify.app/?path=/story/design-system-accordion--multiple) with iOS 13.3 or lower

## Checklist

- [x] [Conventional Commits](https://www.conventionalcommits.org/)
- [x] Storybook

### CODE REVIEW

File: packages/accordion/src/AccordionDetails.tsx [hunk 1]
```diff
@@ -19,18 +19,24 @@ export function AccordionDetails({
...
       if (contentRef.current) {
-        setInnerContentHeight(contentRef.current.scrollHeight)
+        resizeObserver.current.observe(contentRef.current)
+      }
+      return () => {
+        resizeObserver.current?.disconnect()
+      }
+    } else {
+      if (contentRef?.current?.scrollHeight) {
+        setInnerContentHeight(contentRef?.current?.scrollHeight)
```
Comment (eeeeean) [inline packages/accordion/src/AccordionDetails.tsx:38]: This is the fix, if `ResizeObserver` is `undefined` use `setInnerContentHeight` without `ResizeObserver`

Comment (eddkmm) [reply packages/accordion/src/AccordionDetails.tsx:38]: Can we add a comment explaining why we'd need to check if ResizeObserver isn't `'undefined'`

**Review by eddkmm — APPROVED** [review]

**Review by andrii-matviichuk — APPROVED** [review]

File: packages/accordion/src/AccordionDetails.tsx [hunk 2]
```diff
@@ -19,18 +19,24 @@ export function AccordionDetails({
 
   /* istanbul ignore next */
   useEffect(() => {
-    // create the observer only once
-    resizeObserver.current = new ResizeObserver(() => {
+    if (typeof ResizeObserver !== 'undefined') {
```
Comment (huhgawz) [inline packages/accordion/src/AccordionDetails.tsx:22]: nit:
```suggestion
    if ('ResizeObserver' in window) {
```

**Review by huhgawz — APPROVED** [review]

File: packages/accordion/src/AccordionDetails.tsx (hunk 1 above)
Comment (eeeeean) [reply packages/accordion/src/AccordionDetails.tsx:39]: See [76b2fe9](https://github.com/little-spoon-dev/design-system/pull/817/commits/76b2fe934c5e25afb3ac4369b0ca7f70db30a20b)

**Comment by huhgawz** [comment]
Why don't we add a `ResizeObserver` polyfill?

File: packages/accordion/src/AccordionDetails.tsx (hunk 2 above)
Comment (eeeeean) [reply packages/accordion/src/AccordionDetails.tsx]: See [19b038f](https://github.com/little-spoon-dev/design-system/pull/817/commits/19b038f3a785ab3d75c7f3e9db50c30856affc56)

**Comment by eeeeean** [comment]
> Why don't we use a `ResizeObserver` polyfill such as [resize-observer-polyfill](https://www.npmjs.com/package/resize-observer-polyfill) or [resize-observer](https://www.npmjs.com/package/resize-observer)?

@fjallkraken recommended I avoid polyfills - however as far as I can tell the current approach is essentially the same experience for the user.

**Comment by Zaggen** [comment]
> resize-observer

@huhgawz I suggested @eeeeean not to use the polyfill in this case because this only affects a very small portion of our users; it's better to have a degraded experience for them than to increase the bundle size for everybody else.

**Review by Zaggen — APPROVED** [review]

**Review by JenniferBrody — APPROVED** [review]
QA done, looks good!

**Pull request merged**